nautobot/wireless/factory.py

```diff
@@ -26,15 +26,30 @@
 class AccessPointGroupFactory(PrimaryModelFactory):
     class Meta:
         model = models.AccessPointGroup
-        exclude = ("has_description", "has_tenant")
+        exclude = ("has_description", "has_tenant", "has_controller")
 
     name = UniqueFaker("company")
     has_description = NautobotBoolIterator()
     description = factory.Maybe("has_description", factory.Faker("sentence"), "")
-    controller = random_instance(lambda: Controller.objects.filter(controller_device__isnull=True), allow_null=True)
+    has_controller = NautobotBoolIterator()
+    controller = factory.Maybe(
+        "has_controller",
+        random_instance(lambda: Controller.objects.filter(controller_device__isnull=True), allow_null=True),
+        None,
+    )
     has_tenant = NautobotBoolIterator()
     tenant = factory.Maybe("has_tenant", random_instance(Tenant), None)
 
+    @factory.post_generation
+    def devices(self, create, extracted, **kwargs):
+        if create:
+            if extracted:
+                self.devices.set(extracted)
+            else:
+                self.devices.set(
+                    get_random_instances(Device.objects.filter(access_point_group__isnull=True), minimum=1)
+                )
```

Comment:
Does this work? I thought you couldn't set foreign key relationships from the reverse relation

Reply:
That could be why there is way less devices connected. I will investigate.